File: problem-solving/baekjoon/step18/1541_잃어버린_괄호.py
import sys

def greedy_algorithm(formula):
    result = 0
    for i in range(len(formula)):
        formula[i] = sum(list(map(int, formula[i].split('+'))))
        if i == 0:
            result = formula[i]
        else:
            result = result - formula[i]
    return result

if __name__ == "__main__":
    formula = sys.stdin.readline().split('-')
    result = greedy_algorithm(formula)
    print(result)

File: problem-solving/baekjoon/1541_잃어버린_괄호.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	var formula string
	fmt.Fscanln(reader, &formula)
	formulas := strings.Split(formula, "-")
	fmt.Println(greedyAlgorithm(formulas))
}

func greedyAlgorithm(formulas []string) (result int) {
	var values = make([]int, len(formulas))
	for i := 0; i < len(formulas); i++ {
		for _, v := range strings.Split(formulas[i], "+") {
			intVal, _ := strconv.Atoi(v)
			values[i] += intVal
		}
		if i == 0 {
			result = values[i]
		} else {
			result -= values[i]
		}
	}
	return
}
